src/Compilers/CSharp/Portable/Binder/LocalScopeBinder.cs

```diff
@@ -226,6 +226,17 @@ protected ImmutableArray<LocalSymbol> BuildLocals(SyntaxList<StatementSyntax> st
                                 ExpressionVariableFinder.FindExpressionVariables(this, locals, vdecl, localDeclarationBinder);
                             }
 
+                            if (decl.Declaration.Type is ArrayTypeSyntax arrayTypeSyntax)
+                            {
+                                foreach (var rankSpecifier in arrayTypeSyntax.RankSpecifiers)
+                                {
+                                    foreach (var expression in rankSpecifier.Sizes)
+                                    {
+                                        ExpressionVariableFinder.FindExpressionVariables(this, locals, expression, enclosingBinder.GetBinder(expression) ?? enclosingBinder);
```

Comment:
>locals [](start = 95, length = 6)

The order in which locals are added into the list is significant. We are adding these locals after we are adding locals from all the declarators. Is this the right order? Please come up with tests demonstrating that the right order is used (locals shadow each other as expected).